### PR TITLE
Reconfigure eslint to pull from `.gitignore` and per-dir `.eslintignore`

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

Unfortunately we have to do it in a somewhat hacky manner: in eslintrc
we run a function that reads all the ignore files and dumps the results
into eslint's `ignorePatterns` config.

Then we have to do it all over again for any sub-eslintrc that does
`root: true` or that extends any eslintrc that has the hack. Inheritance
is ok, but apparently we don't do that all that often (sigh).

#### Jetpack product discussion

None

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* The files linted by this versus current master on a clean checkout should be the same, other than the added/renamed files.
* The lints reported (for excluded files) should also be the same, other than the two in eslintrc files that are fixed in this PR.
* Do we want to hack the config like this for the convenience of no longer having to copy everything from `.gitignore` to a root-level `.eslintignore`?

### CODE REVIEW

**Comment by anomiex** [comment]
Hmm. Fusion thinks two files from this PR should be synced, but neither actually exist on wpcom. I'm inclined to continue to leave them nonexistent.

File: projects/packages/search/.eslintrc.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
+const loadIgnorePatterns = require( '../../../tools/js-tools/load-eslint-ignore.js' );
+
 module.exports = {
 	// Use root level ESlint configuration.
 	// JavaScript files inside this folder are meant to be transpiled by Webpack.
 	root: true,
 	extends: [ '../../../.eslintrc.react.js' ],
-	ignorePatterns: [ '**/external/*.js' ],
+	ignorePatterns: loadIgnorePatterns( __dirname ),
```
Comment (kraftbj) [inline projects/packages/search/.eslintrc.js:8]: @anomiex Can you comment on this -- this came from the `master` merge. Search added the `external` ignorePattern and you added the `loadIgnorePatterns`.

Should only one remain, both?

**Review by kraftbj — CHANGES_REQUESTED** [review]
One question when trying to resolve the merge conflicts.

File: projects/packages/search/.eslintrc.js (hunk 1 above)
Comment (anomiex) [reply projects/packages/search/.eslintrc.js:8]: The `**/external/*.js` should probably be put into a `projects/packages/search/.eslintignore`

**Review by kraftbj — DISMISSED** [review]

**Comment by kraftbj** [comment]
> Hmm. Fusion thinks two files from this PR should be synced, but neither actually exist on wpcom. I'm inclined to continue to leave them nonexistent.

It's a somewhat known issue that Fusion is over-eager to match files of the same name in different folders when there isn't an exact match. Agreed to force-pass this one.

**Review by kraftbj — APPROVED** [review]

**Pull request merged**